File: TrainingSchedule.cpp
#include "TrainingSchedule.hpp"
#include "ProfileManager.hpp"
#include "FitnessDatabase.hpp"
#include <stdexcept>

// Инициализация статического поля
int TrainingSchedule::totalSchedules = 0;

// Default-конструктор: Инициализирует пустыми значениями
TrainingSchedule::TrainingSchedule() : id(""), ownerUserId(""), startDate(""), endDate(""), sessions({}) {
    totalSchedules++;
}

// Конструктор: Инициализирует все поля
TrainingSchedule::TrainingSchedule(const std::string& id, const std::string& ownerUserId, const std::string& startDate,
    const std::string& endDate, const std::list<TrainingSession>& sessions)
    : id(id), ownerUserId(ownerUserId), startDate(startDate), endDate(endDate), sessions(sessions) {
    totalSchedules++;
}

// Конструктор копирования
TrainingSchedule::TrainingSchedule(const TrainingSchedule& other)
    : id(other.id), ownerUserId(other.ownerUserId), startDate(other.startDate),
    endDate(other.endDate), sessions(other.sessions) {
    totalSchedules++;
}

// Деструктор: Пустой
TrainingSchedule::~TrainingSchedule() {
    totalSchedules--;
}

// Метод: Генерирует план (заглушка, возвращает копию текущего)
TrainingSchedule TrainingSchedule::createPlan(const ProfileManager& user, const FitnessDatabase& library) {
    // Генерация плана
    return *this; // Пример
}

// Метод: Корректировка(заглушка)
void TrainingSchedule::adjustFromFeedback(const std::string& userFeedback) {
    try {
        if (userFeedback.empty()) {
            throw std::invalid_argument("Отзыв не может быть пустым");
        }
        // Корректировка
    }
    catch (const std::exception& e) {
        throw;
    }
}

// Метод: Возвращает предстоящие сессии (заглушка)
std::list<TrainingSession> TrainingSchedule::listNextSessions(int count) const {
    std::list<TrainingSession> result;
    return result;
}

// Перегрузка оператора +
TrainingSchedule TrainingSchedule::operator+(const TrainingSession& session) const {
    TrainingSchedule newSchedule = *this;
    newSchedule.sessions.push_back(session);
    return newSchedule;
}

File: FitnessAppCsharp/TrainingSchedule.cs
using System;
using System.Collections.Generic;

public class TrainingSchedule
{
    // Поля 
    private string id;
    private string ownerUserId;
    private string startDate;
    private string endDate;
    private List<TrainingSession> sessions;

    // Конструктор по умолчанию
    public TrainingSchedule()
    {
        id = "";
        ownerUserId = "";
        startDate = "";
        endDate = "";
        sessions = new List<TrainingSession>();
    }

    // Конструктор с параметрами
    public TrainingSchedule(string id, string ownerUserId, string startDate,
        string endDate, List<TrainingSession> sessions)
    {
        this.id = id;
        this.ownerUserId = ownerUserId;
        this.startDate = startDate;
        this.endDate = endDate;
        this.sessions = new List<TrainingSession>(sessions); 
    }

    // Метод: генерирует новый план (заглушка — возвращает копию текущего)
    public TrainingSchedule CreatePlan(ProfileManager user, FitnessDatabase library)
    {
       
        return new TrainingSchedule(this.id, this.ownerUserId, this.startDate, this.endDate, this.sessions);
    }

    // Метод: корректирует план (заглушка)
    public void AdjustFromFeedback(string userFeedback)
    {
        // Заглушка 
    }

    // Метод: возвращает предстоящие сессии (заглушка)
    public List<TrainingSession> ListNextSessions(int count)
    {
        return new List<TrainingSession>(); 
    }

  
    public string GetId() { return id; }
    public string GetOwnerUserId() { return ownerUserId; }
    public string GetStartDate() { return startDate; }
    public string GetEndDate() { return endDate; }
    public List<TrainingSession> GetSessions() { return sessions; }

   
}